WarOOP.Tests/ArmyTests.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections.Generic;
+using WarOOP.Models;
+using Xunit;
+
+namespace Tests;
+
+public class ArmyTests
+{
+    public static IEnumerable<object[]> TestData =>
+        new List<object[]>
+            {
+                new object[] { typeof(Warrior), 15, typeof(Knight), 7, true },
+                new object[] { typeof(Knight), 10, typeof(Knight), 7, true },
+                new object[] { typeof(Warrior), 5, typeof(Knight), 10, false },
+                new object[] { typeof(Knight), 10, typeof(Knight), 12, false },
+            };
+
+    [Theory]
+    [MemberData(nameof(TestData))]
+    public void ArmyAndArmy_Fight_Correct(Type armyType1, int armyCount1, Type armyType2, int armyCount2, bool expected)
+    {
+        var army1 = new Army();
+        var army2 = new Army();
+        army1.AddUnits(armyType1, armyCount1);
+        army2.AddUnits(armyType2, armyCount2);
+
+        var result = Battle.Fight(army1, army2);
+
+        Assert.Equal(expected, result);
+    }
+
+    [Fact]
+    public void CreateWarrior_WrongType_Exception()
+    {
+        Assert.Throws<Exception>(()=> Warrior.CreateWarrior(typeof(string)));
+    }
+    
+    [Fact]
+    public void Fight_EmptyArmies_Exception()
+    {
+        var army1 = new Army();
+        var army2 = new Army();
+        
+        Assert.Throws<Exception>(()=> Battle.Fight(army1, army2));
+    }
+    
+    [Fact]
+    public void Fight_EmptyArmy_Correct()
```

Comment:
What does _correct_ mean? You need to put here an expected result. In this test expected result is false. So, you can name it _ReturnsFalse_ or just _false_